ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &propagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
 }
 
-// propagator implements a propagator which uses TextMap internally.
+// propagator implements a Propagator that supports TextMap carriers.
 // It propagates the trace and span IDs, as well as the baggage from the
 // context.
-type propagator struct{ cfg *PropagatorConfig }
+type propagator struct {
+	injectors  []textMapPropagator
+	extractors []textMapPropagator
+}
+
+// makeInjectors returns a list of injectors to apply when propagating a
+// context. By default, only the datadog propagation style will be used.
+// If the DD_PROPAGATION_STYLE_INJECT environment variable is set,
+// this can override the default behavior.
+func makeInjectors(cfg *PropagatorConfig) []textMapPropagator {
+	dd := &datadogPropagator{cfg}
+	b3 := &b3Propagator{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_INJECT")
+	if ps == "" {
+		return []textMapPropagator{dd}
+	}
+	styles := propagationStyleList(ps)
```

Comment:
I don't think we need to be overly flexible here. Let's keep things as simple as possible. Only commas should be good enough. Then, this could be:
```go
list := strings.Split(ps, ",")
```
What do you think?

Reply:
I'm fine either way. The existing implementations are either space separated or have a regex to split the value into a list. [java](https://github.com/DataDog/dd-trace-java/blob/18c791633895bfc2d57ca0865f6ca78f480e7f6d/dd-trace-api/src/main/java/datadog/trace/api/Config.java#L103) [cpp](https://github.com/DataDog/dd-opentracing-cpp/blob/ea185e732118bf591b91b350269d99a6a31ffce8/src/tracer_options.cpp#L34-L35)
So this is for consistency with that behavior.

Reply:
I’m all for simplicity, so I’d say having commas is enough. Can we please do that? If not, let’s at least inline this logic. There is no need to factor it out into a function.